linode/manager
# refactor: [M3-8783] - Migrate /volumes to Tanstack Router

## Description 📝
This PR implements our new routing to `/volumes` 🎉 

Beyond replicating functionalities, it also switches to using params for dialogs (modals & drawers) for the whole feature.
While being a tad more involved in terms of development, the benefits are quite great:
- shareable URLs: routed state allows to encode the state of the UI in the URL, making it easy to share specific views or states via a link (ex: link to an detail drawer from another page without complicated state management)
- browser History: changes in routed state are reflected in the browser's history, allowing users to use the back and forward buttons to navigate through different states.
- deep Linking: users can bookmark or directly navigate to a specific state of the application, which is useful for returning to a specific view.
- persistence: the state is preserved across page reloads, as it is stored in the URL.

An overview of how this works
![Untitled-2024-10-31-1553](https://github.com/user-attachments/assets/11bcbe39-c249-43db-bd15-f696db87d39c)

## Changes  🔄
- Implement new routing for any route under (and including) `/volumes`
- deprecate state based drawer/modal display in favor of routes
- make V2 versions of `usePagination` and `userOrder` to use the new router methods
- Unit tests for all the above

## How to test 🧪

### Verification steps
note: you can use `prod-test-029` to test with an account with many volumes

**Try to break things!**

- test **everything** in /volumes
  - pagination: preferences, reload page on existing and new params etc
    - page size
    - page number
  - query volumes via search
  - navigate to all actions and confirm all is working
  - navigate back & forth with browser button
  - try loading a drawer or modal with an invalid ID

## As an Author I have considered 🤔

*Check all that apply*
